nx-X11/programs/Xserver/hw/nxagent/Keystroke.c

```diff
@@ -249,6 +249,11 @@ static void parse_keystroke_file(void)
   char *homefile = "/.nx/config/keystrokes.cfg";
   char *etcfile = "/etc/nxagent/keystrokes.cfg";
 
+  if (nxagentKeystrokeFileParsed)
+    return;
+
+  nxagentKeystrokeFileParsed = True;
```

Comment:
Getting the code design now after reading more of the code context. The nxagentKeystrokeFileParsed is actually more a Boolean for nxagentKeystrokeFileParsingAttempted. It is saying nothing about its success, only that it happened.

We should add a comment about this near the nxagentKeystrokeFileParsed = True assignment, I'd say.